crates/mongodb-agent-common/src/query/groups.rs

```diff
@@ -0,0 +1,162 @@
+use std::{borrow::Cow, collections::BTreeMap};
+
+use indexmap::IndexMap;
+use mongodb::bson::{self, bson};
+use mongodb_support::aggregate::{Accumulator, Pipeline, Selection, SortDocument, Stage};
+use ndc_models::{FieldName, OrderDirection};
+
+use crate::{
+    aggregation_function::AggregationFunction,
+    constants::GROUP_DIMENSIONS_KEY,
+    interface_types::MongoAgentError,
+    mongo_query_plan::{Aggregate, Dimension, GroupOrderBy, GroupOrderByTarget, Grouping},
+};
+
+use super::{aggregates::convert_aggregate_result_type, column_ref::ColumnRef};
+
+type Result<T> = std::result::Result<T, MongoAgentError>;
+
+// TODO: This function can be infallible once ENG-1562 is implemented.
+pub fn pipeline_for_groups(grouping: &Grouping) -> Result<Pipeline> {
+    let group_stage = Stage::Group {
+        key_expression: dimensions_to_expression(&grouping.dimensions).into(),
+        accumulators: accumulators_for_aggregates(&grouping.aggregates)?,
+    };
+
+    // TODO: ENG-1562 This implementation does not fully implement the
```

Comment:
Do we need to do this work at a specific time or this is behind a capability?

Reply:
Oh right, that's behind the query.aggregates.group_by.order capability